### PR TITLE
DB-4628: CIrcle CI and Git Hub Actions Node Upgrade

### PR DESCRIPTION
**Pull request by mitchellmarkoff**
## What changes were made?
- Config files and docs updated for Node version 18
- GitHub actions run on node 18 lts
- CircleCI jobs run on node 18 lts

## Where were the changes made?
workflow config

## How have the changes been tested?
Github actions and CI workflows of this PR show that node 18 was used
## Additional information

Don't forget to [add a changeset](https://github.com/pantheon-systems/decoupled-kit-js#generating-a-changeset) if needed!

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     # Specify the execution environment. You can specify an image from Dockerhub or use one of our Convenience Images from CircleCI's Developer Hub.
     # See: https://circleci.com/docs/2.0/configuration-reference/#docker-machine-macos-windows-executor
     docker:
-      - image: cimg/node:16.14.0
+      - image: cimg/node:18.13.0
```
Comment (CobyPear) [inline .circleci/config.yml:14]: this is probably fine but might be worth using the `node:18lts` tag so we can ensure we're always testing on the latest lts version.

File: README.md [hunk 2]
```diff
@@ -36,7 +36,8 @@ published on `npm`, and to be consumed by the public, and our starter kits.
 To configure the monorepo for development, the following should be installed
 locally:
 
-- Node.js version 16 LTS. We recommend using [nvm](https://github.com/nvm-sh/nvm)
+- Node.js version 18 LTS. We recommend using
+  [nvm](https://github.com/nvm-sh/nvm)
 - [npm](https://docs.npmjs.com/cli/v8/commands/npm) version 8.x.x
```
Comment (CobyPear) [inline README.md:41]: I think npm is on 9.x.x now

**Review by CobyPear — APPROVED** [review]
a few very small tweaks

**Review by CobyPear — CHANGES_REQUESTED** [review]
oops I think I hit 'approve' by mistake

File: .circleci/config.yml (hunk 1 above)
Comment (mitchellmarkoff) [reply .circleci/config.yml]: Good idea, alias for this is just `lts` https://circleci.com/developer/images/image/cimg/node#image-tags

**Review by CobyPear — APPROVED** [review]
LGTM

**Pull request merged**